packages/transitive-overlay/src/util.ts

```diff
@@ -221,6 +242,10 @@ export function itineraryToTransitive(
   newPlaces.push(makeFromToPlace(itin.legs[itin.legs.length - 1].to, "to"));
 
   itin.legs.forEach((leg, idx) => {
+    // Create shallow copy of the leg
+    const mutableLeg = { ...leg };
```

Comment:
After introducing the function `cloneToStop`, it is no longer necessary to create a shallow clone of the leg. A better to do that is to call `cloneToStop` at line 411 by writing:
```js
// Line 411 below
      const toStop = makeStop(cloneToStop(leg), lastCoord);
```